Apps/Examples/Examples/All Examples/Annotations/AnimatedMarkerExample.swift

```diff
@@ -0,0 +1,117 @@
+        label.text = "Tap anywhere on the map"
+
+        mapView.addSubview(label)
+
+        NSLayoutConstraint.activate([
+            label.topAnchor.constraint(equalTo: view.safeAreaLayoutGuide.topAnchor, constant: 32),
+            label.centerXAnchor.constraint(equalTo: view.centerXAnchor)
+        ])
+
+        mapView.mapboxMap.loadStyleURI(.satelliteStreets)
+
+        // add a tap gesture recognizer that will allow the marker to be animated
+        mapView.addGestureRecognizer(UITapGestureRecognizer(target: self, action: #selector(updatePosition(_:))))
+    }
+
+    private func setupExample() {
+        try? mapView.mapboxMap.style.addImage(UIImage(named: "red_marker")!, id: Constants.markerIconId, stretchX: [], stretchY: [])
+
+        // Create a GeoJSON data source.
+        var source = GeoJSONSource()
+        source.data = .feature(Feature(geometry: .point(Point(currentPosition))))
+
+        try? mapView.mapboxMap.style.addSource(source, id: Constants.sourceId)
+
+        // Create a symbol layer
+        var symbolLayer = SymbolLayer(id: "layer-id")
+        symbolLayer.source = Constants.sourceId
+        symbolLayer.iconImage = .constant(.name(Constants.markerIconId))
+        symbolLayer.iconIgnorePlacement = .constant(true)
+        symbolLayer.iconAllowOverlap = .constant(true)
+
+        try? mapView.mapboxMap.style.addLayer(symbolLayer)
+    }
+
+    @objc private func updateFromDisplayLink(displayLink: CADisplayLink) {
+        let animationProgress = (CACurrentMediaTime() - animationStartTimestamp) / Constants.animationDuration
+
+        guard animationProgress <= 1 else {
+            displayLink.invalidate()
+            self.displayLink = nil
+            return
+        }
+
+        // interpolate from origin to destination according to the animation progress
+        let coordinate = CLLocationCoordinate2D(
+            latitude: origin.latitude + animationProgress * (destination.latitude - origin.latitude),
+            longitude: origin.longitude + animationProgress * (destination.longitude - origin.longitude)
+        )
+
+        // update current position
+        self.currentPosition = coordinate
+
+        // update source with the new marker location
+        try? self.mapView.mapboxMap.style.updateGeoJSONSource(withId: Constants.sourceId,
+                                                              geoJSON: .feature(Feature(geometry: .point(Point(coordinate)))))
+
+    }
+
+    @objc private func updatePosition(_ sender: UITapGestureRecognizer) {
+        let newCoordinate = mapView.mapboxMap.coordinate(for: sender.location(in: mapView))
+
+        // save marker's origin and destination to interpolate between them during the animation
+        destination = newCoordinate
+        origin = currentPosition
+        animationStartTimestamp = CACurrentMediaTime()
+
+        // add display link
+        displayLink = CADisplayLink(target: self, selector: #selector(updateFromDisplayLink(displayLink:)))
```

Comment:
I couldn't find a way to animate a marker with built-in tools, so display link it is.

Reply:
yep, that's the only option right now.

look out for the strong reference cycle though — display link keeps a strong ref to its targets

Reply:
Reference cycle is broken after animation is finished, I added early invalidation of display link when view controller is moving away from the screen to prevent updating the map when moving offscreen.